### ISSUE TITLE
Extension unusable in FireFox 40

### ISSUE THREAD

**Issue by StinVec** [jcooley/lingodisco-firefox-addon#1]
With the release of FF40, the extension can no longer function. The icon is completely gone and it cannot be used. I've read that this issue is affecting several add-ons and being sited as being due to Mozilla having removed the old library for loading icons onto the toolbar in Firefox 40.

**Comment by mohancloudworld** [comment]
I have checked the add-on with Firefox-40 on Windows, Mac & Linux and it is working fine. Can you try re-installing the add-on and let me know the result. 

Update: I mistook this issue for another add-on. Sorry, my mistake.

**Comment by StinVec** [comment]
Thank you for responding. I uninstalled the add-on and restarted FireFox just in case. I then reinstalled the add-on and there is still no toolbar icon on any of my toolbars or available in the icon pool during customize. I also went through and disabled other add-ons to see if there may have been some kind of conflict that was causing this add-on to not have an icon, but nothing worked. The add-on list shows it as installed, but it's non-functional for me (nothing in right-click menus, no toolbar icon, nothing in menu bars, nothing translates upon selecting, etc).

I encountered this issue with another add-on upon upgrading to FF40.0 and that author is working to resolve this problem in their add-on, which is why I thought it likely to be the same issue this add-on is having.

I just upgraded to FF40.0.2, but it had no effect on the issue.

Thank you for attempt to help resolve the issue. However, as it seems I am the only one reporting this issue, and your stating that it is working fine for you, I will just use a different translation add-on in the mean time and try again after a while to see if it works again as I really like this add-on.

**Comment by jcooley** [comment]
Thanks for reporting the issue.  The widget UI library that this addon depends on was removed in FF40. It looks like a lot has changed with the addon submission process since the last release of this addon almost 2 years ago. I will see if I can fix it and submit a new version over the next week.

**Comment by StinVec** [comment]
Thank you very much for responding, jcooley! I am very glad to see that you know what is the cause of the issue and that you will work to adjust the add-on to function properly with the new FF version.

Thank you very much for your add-on and continued support to keep it functional!

**Comment by jcooley** [comment]
I updated the addon and submitted it for review. In the meantime it is unofficially available at https://addons.mozilla.org/firefox/downloads/file/338104/lingodisco_inline_translate-0.8.1-fx.xpi?src=devhub Please try it and leave a comment.

Thanks,

James

**Comment by LandisTwo** [comment]
I am having same issue with FF 40 on (opensuse) Linux / KDE. No context menu or button.
I installed above 'unofficial' version and get context menu item for LingoDisco, but nothing happens after clicking (page flashes, maybe). No output '#TranslatoinOutput' box or window.

ps, i get the obvious 'not signed' warning in 'Add-ons Manager', but is enabled.

**Comment by StinVec** [comment]
Thank you for your quick update.

I've tried to install the above 'unofficial' link, but upon attempting to download it I receive a message that states "The add-on could not be downloaded because of a connection failure on GitHub.com". This sounds to me like the file that is linked does not exist any longer.

Also, the above link states the version in the link text as 0.8.1, though the mozilla add-on page has your previous version from 2013 as being 0.9.1, which would imply the above link is to an older version and not to your newly updated version. Am I misunderstanding the version text in the link?

**Comment by LandisTwo** [comment]
#StinVec - i was able to download and install with no problem, an hour or so ago.
I just tried it again, and it is still working. do you have a GitHub account in good standing?

I checked and the version i now have installed is as yours is: 0.8.1 - hmmm

**Comment by LandisTwo** [comment]
I didn't notice, but I have a button in toolbar now.
It translated to first selection, but has not updated to new selections.
I've restarted FF, disabled > enabled addon, selected text > context menu > [click] LingDisco > check dropdown under 'new button' and it's still the very first selection being translated.

**Comment by jcooley** [comment]
Hi Landis,

 I didn't see the issue with new sections you mentioned on news.google.fr and www.libertation.fr when I check on Linux - I will check Windows later today.

 I uploaded a newer version https://addons.mozilla.org/firefox/downloads/file/338202/lingodisco_inline_translate-1.0.1-fx.xpi?src=devhub

  I had to reload the page to be translated to get it to work. I will update the source on github later today too.

Thanks,

James
